checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -192,10 +201,12 @@ def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | s
             key_to_val_changes = {COUNT_KEY: new_value}
         else:
             key_to_val_changes = self._build_key_to_val_changes(new_value, new_key)
+        config_attrs = new_resource.config.get(block_type, {}).get(block_name, {})
         self._pop_foreach_attrs(new_resource.attributes)
-        self._pop_foreach_attrs(new_resource.config.get(block_type, {}).get(block_name, {}))
+        self._pop_foreach_attrs(config_attrs)
         self._update_attributes(new_resource.attributes, key_to_val_changes)
-        self._update_attributes(new_resource.config.get(block_type, {}).get(block_name, {}), key_to_val_changes)
+        foreach_attrs = self._update_attributes(config_attrs, key_to_val_changes)
+        config_attrs['foreach_attrs'] = foreach_attrs
```

Comment:
is this a new attr you are adding to the custom attribtues? meaning like `resource_id_` and `block_name_`
If so, you probably want to add `_` in the end of the name and also add it to `CustomAttributes`.
If not, you can ignore that comment

Reply:
This is a nested attribute, so ill not add it.
But, good point!